packages/protocol/contracts/governance/Election.sol

```diff
@@ -0,0 +1,779 @@
+    return votes.total.total;
+  }
+
+  /**
+   * @notice Returns the list of validator groups eligible to elect validators.
+   * @return The list of validator groups eligible to elect validators.
+   */
+  function getEligibleValidatorGroups() external view returns (address[] memory) {
+    return votes.total.eligible.getKeys();
+  }
+
+  /**
+   * @notice Returns lists of all validator groups and the number of votes they've received.
+   * @return Lists of all validator groups and the number of votes they've received.
+   */
+  function getEligibleValidatorGroupsVoteTotals()
+    external
+    view
+    returns (address[] memory, uint256[] memory)
+  {
+    return votes.total.eligible.getElements();
+  }
+
+  function validatorAddressFromCurrentSet(uint256 index) external view returns (address) {
+    address validatorAddress;
+    assembly {
+      let newCallDataPosition := mload(0x40)
+      mstore(newCallDataPosition, index)
+      let success := staticcall(5000, 0xfa, newCallDataPosition, 32, 0, 0)
+      returndatacopy(add(newCallDataPosition, 64), 0, 32)
+      validatorAddress := mload(add(newCallDataPosition, 64))
+    }
+
+    return validatorAddress;
+  }
+
+  function numberValidatorsInCurrentSet() external view returns (uint256) {
+    uint256 numberValidators;
+    assembly {
+      let success := staticcall(5000, 0xf9, 0, 0, 0, 0)
+      let returnData := mload(0x40)
+      returndatacopy(returnData, 0, 32)
+      numberValidators := mload(returnData)
+    }
+
+    return numberValidators;
+  }
+
+  /**
+   * @notice Returns a list of elected validators with seats allocated to groups via the D'Hondt
+   *   method.
+   * @return The list of elected validators.
+   * @dev See https://en.wikipedia.org/wiki/D%27Hondt_method#Allocation for more information.
+   */
+  function electValidators() external view returns (address[] memory) {
+    // Only members of these validator groups are eligible for election.
+    uint256 maxNumElectionGroups = Math.min(
+      maxElectableValidators,
+      votes.total.eligible.list.numElements
+    );
+    /*
+    uint256 requiredVotes = electabilityThreshold.multiply(
+      FixidityLib.newFixed(votes.total.total)
+    ).fromFixed();
+    */
+    // TODO(asa): Filter by > requiredVotes
+    address[] memory electionGroups = votes.total.eligible.headN(maxNumElectionGroups);
+    uint256[] memory numMembers = getValidators().getGroupsNumMembers(electionGroups);
+    // Holds the number of members elected for each of the eligible validator groups.
+    uint256[] memory numMembersElected = new uint256[](electionGroups.length);
+    uint256 totalNumMembersElected = 0;
+    // Assign a number of seats to each validator group.
+    while (totalNumMembersElected < maxElectableValidators) {
+      uint256 groupIndex = 0;
+      bool memberElected = false;
+      (groupIndex, memberElected) = dHondt(electionGroups, numMembers, numMembersElected);
```

Comment:
not sure, but calling `dHondt()` with 3 arrays; are they pass by value or copied? if the latter seems like a waste

Reply:
Yeah, checked on https://solidity.readthedocs.io/en/v0.5.12/types.html#data-location-and-assignment-behaviour

Not completely sure, but it does tell that memory to memory assignments are by reference. Then, if a function argument is an assigment (it should) is another question.